dashboards-observability/public/components/application_analytics/components/flyout_components/trace_detail_render.tsx

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { EuiText, EuiSpacer, EuiHorizontalRule, EuiCodeBlock } from '@elastic/eui';
+import { ServiceBreakdownPanel } from '../../../trace_analytics/components/traces/service_breakdown_panel';
+import { SpanDetailPanel } from '../../../trace_analytics/components/traces/span_detail_panel';
+import {
+  handleTraceViewRequest,
+  handleServicesPieChartRequest,
+  handlePayloadRequest,
+} from '../../../trace_analytics/requests/traces_request_handler';
+import React, { useEffect, useMemo, useState } from 'react';
+import { HttpStart } from '../../../../../../../src/core/public';
+import { getListItem } from '../../helpers/utils';
+
+type props = {
+  traceId: string;
+  http: HttpStart;
+  openSpanFlyout: (spanId: string) => void;
+};
+
+export const TraceDetailRender = ({ traceId, http, openSpanFlyout }: props) => {
+  const [fields, setFields] = useState<any>({});
+  const [serviceBreakdownData, setServiceBreakdownData] = useState([]);
+  const [payloadData, setPayloadData] = useState('');
+  const [colorMap, setColorMap] = useState({});
+
+  const renderContent = useMemo(() => {
+    if (!traceId) return <></>;
+    const overviewList = [
+      getListItem('Trace ID', traceId),
+      getListItem('Trace group name', fields.trace_group || '-'),
+      getListItem('Latency', fields.latency),
+      getListItem('Last updated', fields.last_updated),
+      getListItem(
+        'Errors',
+        fields.error_count == null ? (
+          <></>
+        ) : fields.error_count > 0 ? (
+          <EuiText color="danger" size="s" style={{ fontWeight: 430 }}>
+            Yes
+          </EuiText>
+        ) : (
+          'No'
+        )
+      ),
+    ];
+
+    return (
+      <>
+        <EuiText size="m">
+          <span className="panel-title">Overview</span>
+        </EuiText>
+        <EuiSpacer size="s" />
+        {overviewList}
+        <EuiSpacer size="xs" />
+        <EuiHorizontalRule margin="s" />
+        <ServiceBreakdownPanel data={serviceBreakdownData} />
+        <EuiSpacer size="xs" />
+        <EuiHorizontalRule margin="s" />
+        <SpanDetailPanel
+          traceId={traceId}
+          http={http}
+          colorMap={colorMap}
+          page="app"
```

Comment:
Do you want to add a new prop for page since I just hardcoded this because I was only calling it from App Analytics?

Reply:
I think I need the same behavior as app for the events flyout. I'll take another look before the next PR.